### ISSUE TITLE
How to finetune openvoice with own dataset

### ISSUE THREAD

**Issue by OswaldoBornemann** [myshell-ai/OpenVoice#172]
How to finetune openvoice with own dataset

**Comment by thanhnew2001** [comment]
Unfortunately, OpenVoice didn't release the training code and data they use for training. I am working with the finetune script, hope it will work

**Comment by thanhnew2001** [comment]
I make the script for fine-tuning here: https://github.com/thanhnew2001/OpenVoice_ONNX

**Comment by vikassnwloodles** [comment]
Hello @thanhnew2001 Kindly help me resolving the following queries:
1. Do we need to use the config.json and checkpoint.pth files from OpenVoice v2’s checkpoint.zip, or should we use different configuration and checkpoint files specifically adapted for the ONNX version?

2. Regarding the dataset structure:
    - What types of audio files should be placed in the training_data folder versus the reference_voices folder?
    - I currently have 2,000 audio files from 5 different speakers. Should I place all 2,000 files in both the training_data and reference_voices directories, or is there a more specific way to partition the data?

Your guidance on these matters would be greatly appreciated to ensure proper setup and alignment with the intended workflow.

Thank you in advance for your time and support.

**Comment by DarkKowalski** [comment]
Thank you @thanhnew2001, I've tried your fork with both OpenVoice and OpenVoiceV2 checkpoints downloaded from huggingface.

I followed your instruction in README.md and got the following error message repeatedly. 

Running on CUDA and my batch size is 1.

```
Original spec shape: torch.Size([1, 513, 1000])
Reshaped spec: torch.Size([1, 1, 513, 1000])
Error in voice_conversion: Expected 2D (unbatched) or 3D (batched) input to conv1d, but got input of size: [1, 1, 513, 1000]
Spec shape: torch.Size([1, 1, 513, 1000])
src_se shape: torch.Size([1, 256, 1])
ref_se shape: torch.Size([1, 256, 1])
Original spec shape: torch.Size([1, 513, 1000])
Reshaped spec: torch.Size([1, 1, 513, 1000])
Error in voice_conversion: Expected 2D (unbatched) or 3D (batched) input to conv1d, but got input of size: [1, 1, 513, 1000]
Spec shape: torch.Size([1, 1, 513, 1000])
src_se shape: torch.Size([1, 256, 1])
ref_se shape: torch.Size([1, 256, 1])
```

**Comment by MuhammadAmeenBohio** [comment]
Hey @DarkKowalski ,
I saw your comment about the repeated error. Just wondering did you manage to resolve it? I'm running into the same issue. Thanks!

**Comment by DarkKowalski** [comment]
> Hey @DarkKowalski ,
> I saw your comment about the repeated error. Just wondering did you manage to resolve it? I'm running into the same issue. Thanks!

No, I think it is not easy to fine-tune the model without weights of pre-trained discriminators.

**Comment by MuhammadAmeenBohio** [comment]
> > Hey [@DarkKowalski](https://github.com/DarkKowalski) ,
> > I saw your comment about the repeated error. Just wondering did you manage to resolve it? I'm running into the same issue. Thanks!
> 
> No, I think it is not easy to fine-tune the model without weights of pre-trained discriminators.

Thanks!!!
